src/test/test-bpf-foreign-programs.c

```diff
@@ -0,0 +1,337 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <fcntl.h>
+#include <linux/bpf_insn.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <unistd.h>
+
+#include "bpf-foreign.h"
+#include "load-fragment.h"
+#include "manager.h"
+#include "process-util.h"
+#include "rlimit-util.h"
+#include "rm-rf.h"
+#include "service.h"
+#include "tests.h"
+#include "unit.h"
+#include "virt.h"
+
+struct Test {
+        const char *option_name;
+        enum bpf_prog_type prog_type;
+        enum bpf_attach_type attach_type;
+        const char *bpffs_path;
+};
+
+typedef struct Test Test;
+
+static const Test single_prog[] = {
+        {
+                .option_name = "BPFProgram",
+                .prog_type = BPF_PROG_TYPE_CGROUP_SKB,
+                .attach_type = BPF_CGROUP_INET_INGRESS,
+                .bpffs_path = "/sys/fs/bpf/trivial-skb",
```

Comment:
@keszybz

This feature requires bpf filesystem to be mounted and `/sys/fs/bpf/` is the most commonly used mount point, it is usually mounted as system starts:
```
mount bpffs /sys/fs/bpf -t bpf
```

`bpf_foreign_supported` check will fail if bpffs is mounted somewhere else
https://github.com/systemd/systemd/pull/13496/commits/329a5dd70791a081db38be6d1a05c833151ef580#diff-11d3c4581343e0c3d561b27621161777ac95427feb2dd615b3c2f59429e80aefR121

To change that:
- bpffs path should be configurable
- test should mount bpffs

Looks like a follow-up?
I'm not aware about possible implications when mounting bpffs as a nested mount in tmpfs, never tried that.

Reply:
I just meant a path like `/sys/fs/bpf/test-bpf-foreign-trivial-skb` instead of `/sys/fs/bpf/trivial-skb`. This should not require any changes except the string? Anyway, it's fine to do this in a followup.